netket/vqs/full_summ/expect.py

```diff
@@ -65,12 +65,39 @@ def expect(vstate: FullSumState, Ô: DiscreteOperator) -> Stats:  # noqa: F811
     return Stats(mean=expval_O, error_of_mean=0.0, variance=variance)
 
 
-@dispatch
-def expect_and_forces(
+@expect_and_grad.dispatch
+def expect_and_grad_fullsum(
+    vstate: FullSumState,
+    Ô: DiscreteOperator,
+    *args,
+    mutable: CollectionFilter = False,
+    use_covariance: Optional[bool] = None,
+    **kwargs,
+) -> tuple[Stats, PyTree]:
+    if use_covariance is None:
+        use_covariance = Ô.is_hermitian
+
+    if use_covariance:
+        # Implementation of expect_and_grad for `use_covariance == True` (due to the Literal[True]
+        # type in the signature).` This case is equivalent to the composition of the
+        # `expect_and_forces` and `force_to_grad` functions.
+        # return expect_and_grad_from_covariance(vstate, Ô, *args, mutable=mutable)
+        Ō, Ō_grad = expect_and_forces(vstate, Ô, *args, mutable=mutable, **kwargs)
+        Ō_grad = force_to_grad(Ō_grad, vstate.parameters)
+        return Ō, Ō_grad
+    else:
+        raise NotImplementedError(
+            "The gradient of non hermitian operators with "
+            "FullSumState is not yet implemented"
+        )
+
+
+@expect_and_forces.dispatch
+def expect_and_forces_fullsum(
     vstate: FullSumState,
     Ô: DiscreteOperator,
     *,
-    mutable: CollectionFilter,
+    mutable: CollectionFilter = False,
```

Comment:
is CollectionFilter documented somewhere?

Reply:
We don't need this annotation in truth. Dian had added it a while back but we could also remove it. It's a flax concept. In the documentation it should link to flax...